typescript/cli/src/commands/core.ts

```diff
@@ -137,6 +136,16 @@ export const read: CommandModuleWithContext<{
     ),
   },
   handler: async ({ context, chain, mailbox, config: configFilePath }) => {
+    if (!mailbox) {
+      const addresses = await context.registry.getChainAddresses(chain);
+      mailbox = addresses?.mailbox;
+      if (!mailbox) {
+        throw new Error(
```

Comment:
@nbayindirli @ltyu was there a util for this pattern of 'if not given, prompt, then fail if still missing'? I thought someone made one but can't quite remember.

Reply:
I think I made this `detectAndConfirmOrPrompt`

Reply:
in this particular case I dont think we want to burden the user with confirming a mailbox address? like if I do `core read --chain ethereum` I want to know the ethereum core info, not be asked to confirm the ethereum mailbox address?